### PR TITLE
Remove everything that is not new from batch/v2alpha1

### LINKED ISSUE #37166: Remove everything that is not new from batch/v2alpha1
Mentioned [in this comment](https://github.com/kubernetes/kubernetes/pull/36673#issuecomment-261375309), the general takeaway is we should get rid of all objects that are not new in `batch/v2alpha1`. That leave us only with CronJobs being part of that API GroupVersion.

### PR DESCRIPTION
**Pull request by soltysh**
Fixes #37166.

@lavalamp you've asked for it 
@erictune this is a prereq for moving CronJobs to beta. I initially planned to put all in one PR, but after I did that I figured out it'll be easier to review separately. ptal 

@kubernetes/api-approvers @kubernetes/sig-api-machinery-pr-reviews ptal

### CODE REVIEW

**Comment by soltysh** [comment]
@k8s-bot bazel test this

**Comment by soltysh** [comment]
@kubernetes/sig-apps-api-reviews ptal as well

**Comment by soltysh** [comment]
Rebased.

**Comment by soltysh** [comment]
Updated bazel build.

**Comment by soltysh** [comment]
We were still using old client in e2e tests, which forced using batch/v2alpha1, but currently client is smart enough to handle those manually. The last commit removes that part.

**Comment by soltysh** [comment]
Rebased.

**Comment by soltysh** [comment]
@k8s-bot kops aws e2e test this

**Comment by soltysh** [comment]
@erictune && @kubernetes/api-approvers ptal

**Comment by smarterclayton** [comment]
API changes LGTM

/approve

**Comment by soltysh** [comment]
I'm waiting with rebase for https://github.com/kubernetes/kubernetes/pull/39683 to land, to save myself one rebase, at least.

**Comment by soltysh** [comment]
Rebased. @janetkuo mind having a look?

**Comment by soltysh** [comment]
@k8s-bot gce etcd3 e2e test this

File: pkg/apis/batch/v2alpha1/defaults_test.go
Comment (janetkuo) [inline pkg/apis/batch/v2alpha1/defaults_test.go]: This is adding new test for CronJob, not removing Job from v2alpha1. This is okay for now, but in the future, I'd recommend separating this as its own commit or PR. This way, it'll be easier to review and manage (e.g. revert changes).

File: pkg/apis/batch/v2alpha1/defaults_test.go
Comment (janetkuo) [inline pkg/apis/batch/v2alpha1/defaults_test.go]: I just realized we didn't specify default `ConcurrencyPolicy` of this field (although it's mentioned in types.go, it's not discoverable for users, see https://kubernetes.io/docs/api-reference/v1.5/#cronjobspec-v2alpha1). We can fix this in a follow-up PR.

File: pkg/apis/batch/v2alpha1/defaults_test.go
Comment (janetkuo) [inline pkg/apis/batch/v2alpha1/defaults_test.go]: maybe "set fields should not be defaulted"?

**Comment by janetkuo** [comment]
Some comments; LGTM otherwise

File: pkg/apis/batch/v2alpha1/defaults_test.go
Comment (soltysh) [inline pkg/apis/batch/v2alpha1/defaults_test.go]: Yeah, I've realized we didn't have one for CronJobs, I should've put that in a separate commit, will do it next time.

**Comment by soltysh** [comment]
@janetkuo thx, will submit followup with suggested fixes.

**Comment by soltysh** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by soltysh** [comment]
I'm removing lgtm, until I get approval for this exception, apparently I misread the information I got.

**Comment by ethernetdan** [comment]
Removing PR from 1.6 milestone since code freeze has begun:
* If this is a bug fix, please label as `kind/bug` and reapply the milestone.
* If this is a feature for 1.6, an [exception](https://github.com/kubernetes/features/blob/master/EXCEPTIONS.md) should be filed for it before 11:00PM PST on 3/2. Note that most exceptions to code freeze are not accepted.
* If this is a feature for 1.7, feel free to add the 1.7 milestone to the PR.

/cc @kubernetes/release-team @kubernetes/kubernetes-release-managers

**Comment by liggitt** [comment]
Pretty sure we need this to avoid shipping API versions we don't intend to. @lavalamp @smarterclayton care to weigh in?

**Comment by soltysh** [comment]
@ethernetdan I did file an exception for it, waiting to hearing back, this is one of the features we've talked about yesterday.

**Comment by smarterclayton** [comment]
We cannot create new API versions by accident - the exception required
would be to not do this.

This is a release blocker to me.

On Mar 4, 2017, at 12:02 AM, Kubernetes Submit Queue <
notifications@github.com> wrote:

@soltysh <https://github.com/soltysh> PR needs rebase

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/41890#issuecomment-284128183>,
or mute the thread

**Comment by smarterclayton** [comment]
I classify this as a serious bug (exposing new types accidentally).

**Comment by soltysh** [comment]
Rebased.

**Comment by soltysh** [comment]
@k8s-bot bazel test this issue #39975

**Review by janetkuo — APPROVED** [review]
Still LGTM. @smarterclayton may have more comments

File: pkg/apis/batch/v2alpha1/defaults_test.go
Comment (smarterclayton) [inline pkg/apis/batch/v2alpha1/defaults_test.go]: @soltysh don't forget this.

**Comment by smarterclayton** [comment]
/lgtm 

as well

**Comment by soltysh** [comment]
The PR was lgtm-ed by both @smarterclayton and @janetkuo, not sure why bot didn't caught it. Adding lgtm label manually.

**Comment by soltysh** [comment]
@k8s-bot verify test this

**Pull request merged**
